lms/static/js/edxnotes/utils/notes_collector.js

```diff
@@ -0,0 +1,87 @@
+(function(define) {
+    'use strict';
+    define(['jquery', 'underscore', 'annotator_1.2.9'], function($, _, Annotator) {
+        var cleanup,
+            renderNotes,
+            fetchNotesWhenReady,
+            storeNotesRequestData,
+            searchRequestsData = [];
+
+        cleanup = function () {
+            searchRequestsData = [];
+        };
+
+        storeNotesRequestData = function (data) {
+            searchRequestsData.push(data);
+            fetchNotesWhenReady();
+        };
+
+        fetchNotesWhenReady = function () {
+            if ($('.edx-notes-wrapper').length != searchRequestsData.length) {
```

Comment:
@andy-armstrong If i understood correctly then count changes means that new HTML problem is added into the unit? I don't think we need to do anything because new problem will not appear until we refresh page and in that case `notes_collector.js` is initialised and everything will be in clean state.